src/spineml/simulator/simulator.cc

```diff
@@ -0,0 +1,820 @@
+            }
+            else if(strcmp(rateDistribution.value(), "poisson") == 0) {
+                return std::unique_ptr<Input::Base>(
+                    new Input::PoissonSpikeRate(m_DT, node, std::move(inputValue),
+                                                targetSize->second, spikeQueuePtr, hostSpikeCount, hostSpikes,
+                                                pushFunc));
+            }
+            else {
+                throw std::runtime_error("Unsupport spike rate distribution '" + std::string(rateDistribution.value()) + "'");
+            }
+        }
+        // Otherwise, create an exact spike-time input
+        else {
+            return std::unique_ptr<Input::Base>(
+                new Input::SpikeTime(m_DT, node, std::move(inputValue),
+                                     targetSize->second, spikeQueuePtr, hostSpikeCount, hostSpikes,
+                                     pushFunc));
+        }
+    }
+    // Otherwise we assume it's an analogue receive port
+    else {
+        // If there is a dictionary of properties for target population
+        auto targetProperties = m_ComponentProperties.find(target);
+        if(targetProperties != m_ComponentProperties.end()) {
+            // If there is a model property object for this port return an analogue input to stimulate it
+            auto portProperty = targetProperties->second.find(port);
+            if(portProperty != targetProperties->second.end()) {
+                return std::unique_ptr<Input::Base>(
+                    new Input::Analogue(m_DT, node, std::move(inputValue),
+                                        portProperty->second.get()));
+            }
+            else {
+                throw std::runtime_error("Port '" + port + "' not found on target '" + target + "'");
+            }
+        }
+        else {
+            throw std::runtime_error("No properties found for '" + target + "'");
+        }
+    }
+}
+//----------------------------------------------------------------------------
+std::unique_ptr<LogOutput::Base> Simulator::createLogOutput(const pugi::xml_node &node,
+                                                            const filesystem::path &logPath,
+                                                            const std::map<std::string, unsigned int> &componentSizes,
+                                                            const std::map<std::string, std::string> &componentURLs,
+                                                            const ComponentEventPorts &componentEventPorts)
+{
+    const unsigned long long numTimeSteps = calcNumTimesteps();
+    // Get name of target
+    std::string target = node.attribute("target").value();
+
+    // Find size of target component
+    auto targetSize = componentSizes.find(target);
+    if(targetSize == componentSizes.end()) {
+        throw std::runtime_error("Cannot find component '" + target + "'");
+    }
```

Comment:
Maybe this var should be called ``targetSizeElem`` or something? It doesn't actually represent the target size